### ISSUE TITLE
No redirection when changing language in the orphan sentences page

### ISSUE THREAD

**Issue by Guybrush88** [Tatoeba/tatoeba2#2286]
Currently, when I'm browsing the pages of the orphan sentences in a given language which contain lots of sentences (for example English: https://tatoeba.org/ita/activities/adopt_sentences/eng), it may happen that I go to a page with a high number (for example: https://tatoeba.org/ita/activities/adopt_sentences/eng?page=5308). Then, I decide to change the language to one with way less orphan sentences (i.e. Italian, which has none), and I get this error message:

> Not Found
> 
> Errore: The requested address '/ita/Activities/adopt_sentences/ita?page=5308' was not found on this server.

Steps to reproduce the error message:
1) go to https://tatoeba.org/ita/Activities/adopt_sentences/
2) select a language with many orphan sentences
3) go to one of the final pages
4) from there, switch to a language with way less orphan sentences.

Expected behavior: I'd expect to be redirected to the last page containing orphan sentences (or the first one, if no orphan sentences exist) in the second language I want to choose

**Comment by AndiPersti** [comment]
> Expected behavior: I'd expect to be redirected to the last page containing orphan sentences (or the first one, if no orphan sentences exist) in the second language I want to choose

Should we really redirect to the last page in that case? You weren't on the last page of the English sentences.

What about the case were the first chosen language has less pages than the second? E.g. you are on page 3 from language A which happens to be the last page. After changing to language B which happens to have 5 pages, should the redirect go to page 3 or the last (i.e. page 5)?

As I see it whenever you change the language, the set of sentences to display changes and has nothing in common with the old one. So I think we should just redirect to the first page whenever the user changes the language (i.e. remove the `page` query parameter).

**Comment by jiru** [comment]
You have a strong point about redirecting to page 1 after a language change. Being redirected to the last page is a behavior that already exists (when trying to reach a page greater than the last page) so I assume that’s the reason Guybrush had this assumption.

**Comment by Guybrush88** [comment]
> Being redirected to the last page is a behavior that already exists (when trying to reach a page greater than the last page) so I assume that’s the reason Guybrush had this assumption.

Indeed, that's my reason, that's why I suggested it, so there could be some consistency through all the pages

**Comment by AndiPersti** [comment]
I've just noticed that the behavior to redirect to page 1 after a language change is already implemented on the "Browse by language" page which uses `ShowAllHelper::displayShowAllInSelect` instead of `CommonModulesHelper::createFilterByLangMod` for the select form. :roll_eyes: 

`CommonModulesHelper::createFilterByLangMod` is used on a few pages:

Page | After language change, redirects to
-|-
[sentence_comments/index](https://tatoeba.org/eng/sentence_comments/index)|same page or last page (if new language has less pages)
[sentences/of_user](https://tatoeba.org/eng/sentences/of_user/alexmarcelo/)|same page or last page (if new language has less pages)
[tags/show_sentences_with_tag](https://tatoeba.org/eng/Tags/show_sentences_with_tag/7)|same page or error (if new language has less pages)
[vocabulary/add_sentences](https://tatoeba.org/eng/Vocabulary/add_sentences)|same page or error (if new language has less pages)
[vocabulary/of](https://tatoeba.org/eng/Vocabulary/of/CK)|same page or error (if new language has less pages)
[activities/adopt_sentences](https://tatoeba.org/eng/Activities/adopt_sentences)|same page or error (if new language has less pages)
[activities/translate_sentences_of](https://tatoeba.org/eng/activities/translate_sentences_of/alexmarcelo)|same page or last page (if new language has less pages)
[contributions/latest](https://tatoeba.org/eng/Contributions/latest)|impossible to get more than one page

Does it make sense to merge both helpers? Except for the last page behavior the only difference I can see is that the `ShowAllHelper` has `Unknown language` at the end and no `All languages` while the `CommonModulesHelper` has `All languages` at the beginning and no `Unknown language`.

**Comment by jiru** [comment]
Yes it makes sense to merge them. As far as I know, they are part of some very really old code that deserves updating.
